File: TPC v2.0 Online Round/c++/Corona Virus.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    int n;
    string s;
    cin >> n >> s;
    
    std::vector<int> mem;
    std::vector<int> mem2;
    int check = 0, count = 1;
    for(int i=0;i<n;i++){
		if(s[i]=='1'){
			if(check%2==0){
				mem.push_back(i);
			}else{
				mem2.push_back(i);
			}
			check++;
		}    	
	}
	if(mem.size() == mem2.size()){
		for(int i=0;i<mem.size()-1;i++){
			cout << mem[i+1] << mem2[i] << endl;
			count = count * (mem[i+1] - mem2[i]);
		}
		cout << count;
	}
	else{
		cout << -1;
	}
}

File: TPC v2.0 Online Round/python/Corona Virus.py
n = int(input())
s = list(input())

mem = []
count = 1
check = 0
mem2 = []
for i in range(n):
    if s[i] == '1':
        if check%2 == 0:
            mem.append(i)
        else:
            mem2.append(i)
        check += 1

if len(mem) == len(mem2):
    for i in range(0,len(mem)-1):
        if mem[i+1] - mem2[i] != 1:
            count *= (mem[i+1] - mem2[i])
    if count == 0:
        print(count+1)
    else:
        print(count)
else:
    print(-1)
